dev/TreeView/TreeViewNode.h

```diff
@@ -107,21 +107,14 @@ class TreeViewNodeVector :
     TreeViewNodeVector(unsigned int capacity);
 
     void SetParent(winrt::TreeViewNode value);
-    void AppendCore(winrt::TreeViewNode const& item);
-    void InsertAtCore(unsigned int index, winrt::TreeViewNode const& item);
-    void SetAtCore(unsigned int index, winrt::TreeViewNode const& item);   
-    void RemoveAtCore(unsigned int index);
-    void RemoveAtEndCore();
-    void ReplaceAllCore(winrt::array_view<winrt::TreeViewNode const> values);
-    void ClearCore(); 
-
-    void Append(winrt::TreeViewNode const& item);   
-    void InsertAt(unsigned int index, winrt::TreeViewNode const& item);
-    void SetAt(unsigned int index, winrt::TreeViewNode const& item);   
-    void RemoveAt(unsigned int index);    
-    void RemoveAtEnd();
+
+    void Append(winrt::TreeViewNode const& item, bool updateItemsSource = true);   
```

Comment:
I don't suggest you add flags here because Append/InsertAt/SetAt is a common implementation of Vector and should match with the signature of Vector.
Create new functions like AppendAndUpdateItemsSource.

Reply:
> I don't suggest you add flags here because Append/InsertAt/SetAt is a common implementation of Vector and should match with the signature of Vector.
> Create new functions like AppendAndUpdateItemsSource.

I think the intention here is to auto update ItemsSource when TreeViewNode changes and always keep them in sync. So we want the normal Append/InsertAt/SetAt function calls to updateItemsSource, we probably don't want to replace them with a special AppendAndUpdateItemsSource function.

_Not updating_ the ItemsSource is actually the special case, we only need that in ItemsSourceChanged callbacks. So I think it's fine to keep a parameter with default value, the caller should use the Append/InsertAt/SetAt functions just like usual and don't need to worry about the extra parameter in most cases.